sktime/forecasting/compose/_pipeline.py

```diff
@@ -1423,6 +1446,16 @@ def _get_forecaster_X_prediction(self, X=None, fh=None, method="predict"):
 
         return X_pred
 
+    def _get_X_for_fcX(self, X):
+        """Shorthand to obtain X for forecaster_X, depending on parameters."""
+        if X is None or self.forecaster_X_exogeneous != "complement":
+            return None
+        if self.forecaster_X_exogeneous == "complement":
+            X_for_fcX = X.drop(columns=self.columns)
+            if X_for_fcX.shape[1] < 1:
+                return None
```

Comment:
Clarification: what does this check do? to check if complement of `self.columns` is null set?

Reply:
yes, it checks for the edge case that the user provides all the columns as `columns`.